### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by thorncp**
Copies structure, supporting files, etc from hound-jscs with changes for
the mdast-lint api.

### CODE REVIEW

File: lib/linter.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+function parseConfig(config) {
+  if (!config) {
+    config = "{}";
+  }
+
+  try {
+    return JSON.parse(config);
+  } catch (exception) {
+    console.log("Invalid mdast-lint configuration:");
```
Comment (thorncp) [inline lib/linter.js]: This was fixed on other lines.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+var Redis = require("redis");
+var Resque = require("node-resque");
+var Linter = require("./lib/linter");
```
Comment (teoljungberg) [inline index.js]: Is there a way to add `.` or `lib` to the load-path?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+var Queue = require("./lib/queue");
+
+var redis = Redis.createClient(
+  process.env.REDIS_URL || "redis://localhost:6379"
```
Comment (teoljungberg) [inline index.js]: Would it make more sense to store the default redis url in the environment somewhere?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+var outbound = new Queue({
+  redis: redis,
+  queueName: "high",
+  jobName: "CompletedFileReviewJob",
+});
+
+var linter = new Linter(outbound);
+
+var worker = new Resque.worker({
```
Comment (teoljungberg) [inline index.js]: To me, it's not obvious the `Resque.worker` takes two arguments. What do you think about re-formatting the arguments like this to make it clearer?

```javascript
var worker = new Resque.worker(
  {
    connection: { redis: redis },
    queues: ["mdast_review"],
  },
  {
    "MdastReviewJob": linter.lint.bind(linter),
  },
);
```

File: lib/queue.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+Queue.prototype.enqueue = function(job) {
+  var resque = this.resque;
+  var queueName = this.queueName;
+  var jobName = this.jobName;
+
+  return new RSVP.Promise(function(resolve, reject) {
+    resque.connect(function() {
+      resque.enqueue(queueName, jobName, job, function(error, toRun) {
+        if (error) {
+          reject(error);
```
Comment (teoljungberg) [inline lib/queue.js]: What are the promises used for?

File: index.js (hunk 2 above)
Comment (thorncp) [inline index.js]: I have no idea.

File: index.js (hunk 3 above)
Comment (thorncp) [inline index.js]: Probably? I think it would make sense to set dotenv and only have the default fo dev/test, but I'd rather be a separate PR.

File: index.js (hunk 4 above)
Comment (thorncp) [inline index.js]: Makes sense to me. 7b137054a9061075c421430b6df5f01ad3a23a56

File: lib/queue.js (hunk 5 above)
Comment (thorncp) [inline lib/queue.js]: I don't know.

File: index.js (hunk 3 above)
Comment (teoljungberg) [inline index.js]: Makes sense to me

File: lib/queue.js (hunk 5 above)
Comment (teoljungberg) [inline lib/queue.js]: Looping in @seanpdoyle here

Comment (seanpdoyle) [inline lib/queue.js]: @teoljungberg they're an alternative to passing around callbacks.

For instance, it makes [tests](https://github.com/thoughtbot/hound-mdast/blob/ct-initial-implementation/tests/queue-test.js#L22-L27) clearer/easier to synchronize.

Comment (teoljungberg) [inline lib/queue.js]: Makes sense, thanks for explaining!
